Bump JDK version to 21 at funqy related code

yuokada opened the pull request:
Follow up #39822 / Close #45615

yuokada committed:
Bump JDK runtime to 17

gsmet commented on extensions/funqy/funqy-amazon-lambda/maven-archetype/src/main/resources/archetype-resources/sam.jvm.yaml:
```diff
@@ -34,8 +34,8 @@ AWSTemplateFormatVersion: '2010-09-09'
       Properties:
         Handler: io.quarkus.funqy.lambda.FunqyStreamHandler::handleRequest
-        Runtime: java8
+        Runtime: java17
```
Thanks for the PR. Do they have a `java21` runtime? Because if so, I think we should use it by default.

yuokada committed:
amazon-lambda, funqy-amazon-lambda: update AWS Lambda templates to Java 21

- Gradle archetype: set source/targetCompatibility to 21\n- SAM template: switch runtime to java21\n\nRationale: align with JDK 21 baseline and AWS Java 21 runtime.

yuokada committed:
Bump JDK version to 21 as an aws lambda runtime in aws-lambda related extensions
